src/main/webapp/script.js

```diff
@@ -533,39 +537,96 @@ function hideDiffs(cy, elems, deletedNodes, deletedEdges, addedNodes, addedEdges
+ */
+function searchAndHighlight(cy, type, searchFunction) {
+  resetElements(cy);
+  let errorText = "";
+  const query = document.getElementById(type + '-search').value;
+  let result;
+  if (query !== "") {
+    result = searchFunction(cy, query);
+    if (result) {
+      highlightElements(cy, result);
+    } else {
+      errorText = type + " does not exist.";
+    }
+  }
+  document.getElementById(type + '-error').innerText = errorText;
+  return result;
+}
+
+/**
+ * Finds specific node from query and zooms in on it.
+ * Returns node if it exists
  */
 function searchNode(cy, query) {
-  // reset nodes to default color
+  let target = cy.$id(query);
+  if (target.length != 0) {
+    cy.fit(target, 50);
+    return target;
+  }
+}
+/**
+ * Constructs list of nodes that contain specified token
+ * and zooms in.
+ * Returns list of nodes that contain token
+ */
+function searchToken(cy, query) {
+  let target = cy.collection();
   cy.nodes().forEach(node => {
-    node.style('background-color', 'blue');
-    node.style('opacity', '1')
+    if (node.data().tokens.includes(query)) {
+      target = target.add(node);
+    }
   });
-  const target = findNodeInGraph(cy, query);
-  if (target) {
-    cy.nodes().forEach(node => node.style('opacity', '0.25'));
-    target.style('background-color', 'olive');
-    target.style('opacity', '1');
-    cy.fit(target, 50);
-    return true;
-  } else {
-    // fits all nodes on screen
-    cy.fit(cy.nodes(), 50);
-    return false;
+  if (target.length > 0) {
+    const showNode = target[0][0];
+    showNode.tip.show();
+    return target;
   }
 }
 
 /**
- * Finds element in cy graph by id
+ * Highlights collection of nodes and edges
  */
-function findNodeInGraph(cy, id) {
-  if (id.length != 0) {
-    const target = cy.$('#' + id);
-    if (target.length != 0) {
-      return target;
+function highlightElements(cy, target) {
+  cy.nodes().forEach(node => node.toggleClass('non-highlighted', true));
+
+  // highlight desired nodes
+  target.forEach(node => {
+    node.style('border-width', '4px');
+    node.toggleClass('non-highlighted', false);
+  });
+  cy.fit(target[0], 50);
+  document.getElementById('num-selected').innerText = "Number of nodes selected: " + target.length;
+
+  // highlight adjacent edges
+  target.connectedEdges().forEach(edge => {
+    edge.style('line-style', 'dashed');
+    edge.style('z-index', '2');
+  });
+}
+
+/**
+ * Resets highlighted elements to default state
+ */
+function resetElements(cy) {
+  // reset node borders and opacity
+  cy.nodes().forEach(node => {
+    node.style('border-width', '0px');
+    // only change opacity of nodes that were changed
+    // because of highlighting (leave nodes changed due to
+    // mutation alone)
+    if (node.hasClass('non-highlighted')) {
+      node.toggleClass('non-highlighted', false);
     }
-  }
-  return null;
+  });
+
+  // reset edge style
+  cy.edges().forEach(edge => edge.style('line-style', 'solid'));
```

Comment:
Same for edges, you could just define a 'regular' edge style and a 'highlighted' edge style and use that here?